pkg/workceptor/kubernetes.go

```diff
@@ -103,12 +106,40 @@ func (kw *kubeUnit) createPod(env map[string]string) error {
 	if err != nil {
 		return err
 	}
-	pod := &corev1.Pod{
-		ObjectMeta: metav1.ObjectMeta{
-			GenerateName: kw.namePrefix,
-			Namespace:    ked.KubeNamespace,
-		},
-		Spec: corev1.PodSpec{
+	spec := corev1.PodSpec{}
+	if ked.KubePodSpec != "" {
+		reader := bytes.NewReader([]byte(ked.KubePodSpec))
+		decoder := yaml.NewYAMLOrJSONDecoder(reader, 1024)
+		err := decoder.Decode(&spec)
+		if err != nil {
+			return err
+		}
+		foundWorker := false
+		for _, container := range spec.Containers {
+			if container.Name == "worker" {
+				if !container.Stdin {
+					container.Stdin = true
+				}
+				if !container.StdinOnce {
+					container.StdinOnce = true
+				}
+				foundWorker = true
+				break
+			}
+		}
+		if !foundWorker {
+			return fmt.Errorf("At least one container must be named worker")
+		}
+		if spec.RestartPolicy != corev1.RestartPolicyNever {
+			spec.RestartPolicy = corev1.RestartPolicyNever
+		}
+		kw.UpdateFullStatus(func(status *StatusFileData) {
+			status.ExtraData.(*kubeExtraData).Image = ""
```

Comment:
Aren't we already guaranteeing these are empty with the error checking on submission?

Reply:
these will contain the image/command strings from the `worker-kubernetes` definition

Reply:
Hmm ... I wonder if we should make this an error as well?

And the more I think about it, the more I think there ought to be an option for a statically configured pod spec.  To avoid the yaml-inside-yaml marshalling issue, we could have the config option take a filename that contains the pod spec yaml.